watch/watch.go

```diff
@@ -52,13 +65,21 @@ func (q *Queue) Watch() (eventq chan events.Event, cancel func()) {
 	return q.CallbackWatch(nil)
 }
 
+// WatchWithCtx returns a channel where all items published to the queue will
+// be received. The channel will be closed when the provided context is
+// cancelled.
+func (q *Queue) WatchWithCtx(ctx context.Context) (eventq chan events.Event) {
```

Comment:
grpc has a variant of `Dial` which takes a context that they call `DialContext`, so `WatchContext`?